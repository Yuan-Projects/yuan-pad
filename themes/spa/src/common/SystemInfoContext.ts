import { createContext, useContext } from 'react';

export const SystemInfoContext = createContext(null);

export const SystemInfoDispatchContext = createContext<any>(null);

export function useSystemInfo() {
  return useContext(SystemInfoContext);
}

export function useSystemInfoDispatch() {
  return useContext(SystemInfoDispatchContext);
}
