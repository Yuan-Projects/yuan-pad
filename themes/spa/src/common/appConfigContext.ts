import React from "react";
import type { IConfigParams } from '../common/types';

export const initialState: IConfigParams = {
  board_name: '',
  site_close: 0,
  close_reason: '',
  admin_email: '',
  copyright_info: '',
  valid_code_open: '0',
  page_on: 0,
  num_perpage: 0,
  theme: '',
  admin: '',
  lang: '',
  timezone: '0',
  filter_type: 1,
  allowed_tags: '',
  dateformat: "",
  filter_words: "",
}
const AppConfigContext = React.createContext(initialState);

export default AppConfigContext;
