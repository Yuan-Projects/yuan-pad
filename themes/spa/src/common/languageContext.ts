import React from "react";
import type { ITranslationData } from "./types";

const initialState: ITranslationData = {
  DEL_SINGLEUSER_CONFIRM: "",
  "DEL_REPLY_CONFIRM": "",
  "DEL_COMMENT_CONFIRM": "",
  "DEL_SELECTEDUSERS_CONFIRM": "",
  "DEL_SELECTEDCOMMENTS_CONFIRM": "",
  "UPDATE_IPLIST_CONFIRM": "",
  "WELCOME_SYS": "",
  "THANKS": "",
  "STATS_INFO": "",
  "NUM_POSTS": "",
  "NUM_REPLY": "",
  "MP_VERSION": "",
  "SYS_INFO": "",
  "PHP_VERSION": "",
  "GD_VERSION": "",
  "SAFE_MODE": "",
  "SYS_CONF": "",
  "BOARD_NAME": "",
  "CLOSE_BOARD": "",
  "CLOSE_REASON": "",
  "COPY_INFO": "",
  "SYS_THEME": "",
  "TIMEZONE": "",
  "POST_CONF": "",
  "FILTER_WORDS": "",
  "ENABLE_CAPTCHA": "",
  "ENABLE_PAGE": "",
  "PAGINATION_TIP": "",
  "POST_PERPAGE": "",
  "ADMIN_CONF": "",
  "CHANGE_PWD": "",
  "PWD_TIP": "",
  "RESET": "",
  "SELECT": "",
  "OPERATION": "",
  "REPLY": "",
  "UPDATE": "",
  "BAN": "",
  "DELETE": "",
  "YOU_REPLIED": "",
  "DELETE_THIS_REPLY": "",
  "CHECK_ALL": "",
  "CHECK_NONE": "",
  "CHECK_INVERT": "",
  "DELETE_CHECKED": "",
  "DELETE_ALL": "",
  "DELETE_ALL_REPLY": "",
  "BACKUP": "",
  "BAD_IP": "",
  "CANCEL": "",
  "TIPS": "",
  "TZ_ZONES": {},
  "WELCOME": "",
  "WELCOME_POST": "",
  "NICKNAME": "",
  "MESSAGE": "",
  "TIME": "",
  "PAGE_NAV": "",
  "ADMIN_REPLIED": "",
  "CLICK_POST": "",
  "CONTENT": "",
  "SUBMIT": "",
  "POST_SHORTCUT": "",
  "ADMIN_EMAIL": "",
  "ACP": "",
  "ACP_INDEX": "",
  "VALIDATE_CODE": "",
  "ACP_LOGIN": "",
  "BACK": "",
  "LOGIN": "",
  "ADMIN_NAME": "",
  "ADMIN_PWD": "",
  "HOME": "",
  "LOGOUT": "",
  "ACP_OVERVIEW": "",
  "ACP_CONFSET": "",
  "ACP_MANAGE_POST": "",
  "ACP_MANAGE_IP": "",
  "LANG": "",
  "YES": "",
  "NO": "",
  "LOGIN_ERROR": "",
  "USERNAME_NOT_EMPTY": "",
  "USERNAME_NOT_AVAILABLE": "",
  "PWD_NOT_EMPTY": "",
  "USERNAME_TOO_SHORT": "",
  "MESSAGE_NOT_EMPTY": "",
  "CAPTCHA_NOT_EMPTY": "",
  "EMAIL_INVALID": "",
  "POST_OK": "",
  "CLICK_TO_REFRESH": "",
  "CAPTCHA": "",
  "DEL_ALL_CONFIRM": "",
  "DEL_ALL_REPLY_CONFIRM": "",
  "WHERE_AM_I": "",
  "QUERY_ERROR": "",
  "REPLY_EMPTY": "",
  "BACKUP_NOTSUPPORT": "",
  "BACKUP_TYPE_NOTSUPPORT": "",
  "FILL_NOT_COMPLETE": "",
  "WORDS_TOO_LONG": "",
  "CAPTCHA_WRONG": "",
  "UNKNOWN": "",
  "NOT_SUPPORT": "",
  "LOGIN_DENIED": "",
  "USERNAME": "",
  "PASSWORD": "",
  "EMAIL": "",
  "REGISTER": "",
  "USERUPDATEFAILED": "",
  "OFF_LINE_MODE": "",
  "SENDING": "",
  "THEMES_DIR_NOTEXISTS": "",
  "SMILEY_DIR_NOTEXISTS": "",
  "CONFIG_FILE_NOTEXISTS": "",
  "CONFIG_FILE_NOTWRITABLE": "",
  "SITENAME_ERROR": "",
  "SITESTATUS_ERROR": "",
  "SITECLOSEREASON_ERROR": "",
  "ADMINEMAIL_ERROR": "",
  "COPYRIGHT_ERROR": "",
  "BADWORDS_ERROR": "",
  "CAPTCHASTATUS_ERROR": "",
  "PAGINATIONSTATUS_ERROR": "",
  "TIMEZONE_ERROR": "",
  "PAGINATION_PARAMETER_ERROR": "",
  "THEME_ERROR": "",
  "ADMINNAME_ERROR": "",
  "ADMINPASS_ERROR": "",
  "LANGUAGE_ERROR": "",
  "INSTALL_PANEL": "",
  "INSTALL_MP": "",
  "ADMIN_USERNAME": "",
  "ADMIN_USERNAME_MIN": "",
  "ADMIN_PASSWORD": "",
  "DB_TYPE_SELECT": "",
  "DB_TYPE": "",
  "DB_NAME": "",
  "DB_USER": "",
  "DB_PWD": "",
  "DB_HOST": "",
  "TB_PREFIX": "",
  "INSTALL": "",
  "FINISHED": "",
  "RETRY": "",
  "DATADIR_NOT_WRITABLE": "",
  "DB_CONNECT_ERROR": "",
  "INSTALL_NEED_HELP": "",
  "INSTALL_AGREEMENT": "",
  "GD_DISABLED_NOTICE": "",
  "SEARCH_NOTFOUND": "",
  "SEARCH_FOUND": "",
  "ONLY_POST": "",
  "NO_SEARCH_PARAM": "",
  "API_REQUEST_ERROR": "",
  "LOGIN_REQUIRED": "",
  "PARAM_ERROR": "",
  "USER_ADMIN": "",
  "DEL_ALLUSER_CONFIRM": "",
  "ALLOWED_TAGS": "",
  "FILTER_HTML_TAGS": "",
  "STRIP_DISALLOWED_TAGS": "",
  "ESCAPE_ALL_TAGS": "",
  "ALLOWED_HTML_TAGS": "",
  "USER_NOT_EXISTS": "",
};
const LanguageContext = React.createContext(initialState);

export default LanguageContext;
