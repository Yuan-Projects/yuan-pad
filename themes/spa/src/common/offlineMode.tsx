import { useContext } from 'react';
import LanguageContext from '../common/languageContext';
import AppConfigContext from './appConfigContext';
function Offline() {
  const appConfig = useContext(AppConfigContext);
  const lang = useContext(LanguageContext);
  return (
    <p
      className="bg-warning"
      style={{
        display: appConfig.site_close == 1 ? 'block' : 'none',
      }}
    >
      {lang.OFFLINE_WARNING}
    </p>
  );
}

export default Offline;
