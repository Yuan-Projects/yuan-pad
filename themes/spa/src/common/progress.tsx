import Modal from 'react-modal';
import customStyles from './ModalStyles';

function Progress({ loadingModalIsOpen }: { loadingModalIsOpen: boolean }) {
  return (
    <Modal ariaHideApp={false} isOpen={loadingModalIsOpen} style={customStyles}>
      <progress></progress>
    </Modal>
  );
}

export default Progress;
