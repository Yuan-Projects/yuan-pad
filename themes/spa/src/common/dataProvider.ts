import axios from 'axios';
import qs from 'qs';

import { SignInData, UpdateCommentObj } from '../common/types';

function banIP(ip: string) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({ip: ip}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'badip', action: 'create'
    }
  });
}

function deleteMultiIPs(ips: string[]) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({select_ip: ips}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'badip', action: 'update'
    }
  });
}

function signIn(credentials: SignInData) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(credentials),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'login'
    }
  });
}

function getUserInfo() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'user', action: 'getUserInfo'
    }
  });
}

function getAppConfig() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'config', action: 'show'
    }
  });
}

function signOut() {
  return axios({
    method: 'post',
    url: 'index.php',
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'logout'
    }
  });
}

function updateUser(userData: object) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(userData),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'update'
    }
  });
}

function signUp(userData: object) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(userData),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'create'
    }
  });
}

function loadCommentsFromServer(pageId: number | string) {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'post', action: 'list', page: pageId
    }
  });
}

function loadAllCommentsFromServer() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'post', action: 'all'
    }
  });
}

function search(keyword: string) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({s:keyword}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'search'
    }
  });
}

function createPost(comment: any) {
  comment.ajax = true;
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(comment),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'post', action: 'create'
    }
  });
}

function getAppConfigACP() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'config', action: 'showAll'
    }
  });
}

function updateSiteConfig(configObj: object) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(configObj),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'config', action: 'update'
    }
  });
}

function getAllUsers() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'user', action: 'list'
    }
  });
}

function deleteAllReplies() {
  return axios({
    method: 'post',
    url: 'index.php',
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'reply', action: 'deleteAll'
    }
  });
}

function deleteAllComments() {
  return axios({
    method: 'post',
    url: 'index.php',
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'post', action: 'deleteAll'
    }
  });
}

function deleteAllUsers() {
  return axios({
    method: 'post',
    url: 'index.php',
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'deleteAll'
    }
  });
}

function deleteComment(commentId: number, reply: string) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({mid: commentId, reply: reply}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'post', action: 'delete'
    }
  });
}

function deleteMutiComments(dataObj: number[]) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({select_mid: dataObj}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'post', action: 'delete_multi_messages'
    }
  });
}

function deleteReply(commentId: number) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({mid: commentId}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'reply', action: 'delete'
    }
  });
}

function deleteUser(uid: number) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({uid: uid}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'delete'
    }
  });
}

function deleteMutiUsers(uids: Array<string | number>) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({select_uid: uids}),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'user', action: 'delete_multi'
    }
  });
}

/**
 * Get cookie value by a specific name.
 * http://stackoverflow.com/a/15724300
 */
function getCookie(name: string) {
  const value = "; " + document.cookie;
  const parts = value.split("; " + name + "=");
  if (parts.length == 2) {
    return parts.pop()?.split(";").shift();
  } else {
    return null;
  }
}

function getTranslations() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'config', action: 'getTranslations'
    }
  });
}

function getSystemInformation() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'site', action: 'getSystemInformation'
    }
  });
}

function createReply(replyData: any) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({
      mid: replyData.pid,
      content: replyData.content
    }),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'reply', action: 'create'
    }
  });
}

function updateReply(replyData: any) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify({
      mid: replyData.pid,
      content: replyData.content
    }),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'reply', action: 'update'
    }
  });
}

function updateComment(commentData: UpdateCommentObj) {
  return axios({
    method: 'post',
    url: 'index.php',
    data: qs.stringify(commentData),
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''
    },
    params: {
      controller: 'post', action: 'update'
    }
  });
}

function getIPBlackList() {
  return axios({
    method: 'get',
    url: 'index.php',
    headers: {'RequestVerificationToken': getCookie('CSRF-TOKEN') || ''},
    params: {
      controller: 'badip', action: 'list'
    }
  });
}

export default {
  banIP,
  createPost,
  createReply,
  deleteAllComments,
  deleteAllReplies,
  deleteAllUsers,
  deleteComment,
  deleteMutiComments,
  deleteReply,
  deleteUser,
  deleteMutiUsers,
  deleteMultiIPs,
  getIPBlackList, 
  getAppConfig,
  getAppConfigACP,
  getAllUsers,
  getSystemInformation,
  getTranslations,
  signIn,
  signOut,
  signUp,
  updateComment,
  updateReply,
  updateUser,
  loadAllCommentsFromServer,
  loadCommentsFromServer,
  getUserInfo,
  search,
  updateSiteConfig
};