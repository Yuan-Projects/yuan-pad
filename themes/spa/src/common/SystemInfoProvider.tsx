import { useReducer } from 'react';
import { SystemInfoContext, SystemInfoDispatchContext } from './SystemInfoContext';
import dataProvider from './dataProvider';

export default function SystemInfoProvider({ children }) {
  const [systemInfo, dispatchBase] = useReducer(systemInfoReducer, initialState);
  const dispatch = dispatchMiddleware(dispatchBase);

  return (
    <SystemInfoContext.Provider value={systemInfo}>
      <SystemInfoDispatchContext.Provider value={dispatch}>{children}</SystemInfoDispatchContext.Provider>
    </SystemInfoContext.Provider>
  );
}

export function dispatchMiddleware(dispatch) {
  return async (action) => {
    switch (action.type) {
      case 'loaded': {
        const res = await dataProvider.getSystemInformation();
        if (res.status === 200 && res.data.statusCode === 200) {
          dispatch({
            type: 'loaded',
            data: res.data.response,
          });
        } else {
          //alert(res.data.statusText);
        }
        break;
      }
      default:
        return dispatch(action);
    }
  };
}

function systemInfoReducer(systemInfo, action) {
  switch (action.type) {
    case 'loaded': {
      return {
        ...systemInfo,
        ...action.data,
      };
    }
    default: {
      throw Error('Unknown action: ' + action.type);
    }
  }
}

const initialState = {
  commentsTotal: 0,
  repliesTotal: 0,
  appVersion: '',
  phpVersion: '',
  gd_loaded: false,
  gdVersion: '',
  registerGlobals: '',
  magicQuotesGPC: '',
  zipSupport: '',
  themes: {},
  timezones: {},
  languages: {},
  dateFormates: {},
};
