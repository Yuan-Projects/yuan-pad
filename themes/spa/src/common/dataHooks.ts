import useSWR from 'swr';
import useSWRMutation from 'swr/mutation'
import dataProvider, { fetchByFunctionName } from './dataProvider';
import { initialState as userInitalState } from './userContext';
import { initialState as appConfigInitalState } from './appConfigContext';
import { initialState as languageInitalState } from './languageContext';
import { initialState as systemInfoInitialState } from './SystemInfoProvider'

const commentListInitalState = {
  total: 0,
  comments: []
};

const swrOptions = {
  revalidateOnFocus: false,
};

export function useUser() {
  const { data, error, isLoading } = useWrappedSWR('getUserInfo', { fallbackData:userInitalState });
 
  return {
    user: data,
    isLoading,
    isError: error
  }
}

export function useAllUsers() {
  return useWrappedSWR('getAllUsers', {fallbackData: []});
}

export function useIPBlackList() {
  return useWrappedSWR('getIPBlackList', {fallbackData: []});
}

export function useSystemInformation() {
  return useWrappedSWR('getSystemInformation', { fallbackData: systemInfoInitialState});
}

export function useAllComments() {
  return useWrappedSWR('loadAllCommentsFromServer', { fallbackData: {}});
}

function useWrappedSWR(key: any, swrExtraOptions = {}) {
  return useSWR(key, fetchByFunctionName, {
    ...swrOptions,
    ...swrExtraOptions,
  });
}

async function sendRequest(url: any, { arg } : { arg: any}) {
  // @ts-expect-error Expected
  return dataProvider[url](arg).then((res) => {
    if (res.data.statusCode !== 200) {
      throw res.data.response;
    }
    return res.data.response;
  });
}

export function useDeleteAllUsers() {
  return useSWRMutation('deleteAllUsers', sendRequest /* options */);
}

export function useDeleteMutiUsers() {
  return useSWRMutation('deleteMutiUsers', sendRequest /* options */);
}


export function useDeleteAllComments() {
  return useSWRMutation('deleteAllComments', sendRequest /* options */);
}

export function useDeleteAllReplies() {
  return useSWRMutation('deleteAllReplies', sendRequest /* options */);
}

export function useDeleteMutiComments() {
  return useSWRMutation('deleteMutiComments', sendRequest /* options */);
}

export function useDeleteComment() {
  return useSWRMutation('deleteComment', sendRequest /* options */);
}

export function useUpdateReply() {
  return useSWRMutation('updateReply', sendRequest /* options */);
}

export function useCreateReply() {
  return useSWRMutation('createReply', sendRequest /* options */);
}

export function useDeleteUser() {
  return useSWRMutation('deleteUser', sendRequest /* options */);
}

export function useUpdateComment() {
  return useSWRMutation('updateComment', sendRequest /* options */);
}

export function useDeleteReply() {
  return useSWRMutation('deleteReply', sendRequest /* options */);
}

export function useBanIP() {
  return useSWRMutation('banIP', sendRequest /* options */);
}

export function useDeleteMultiIPs() {
  return useSWRMutation('deleteMultiIPs', sendRequest /* options */);
}

export function useUpdateConfig() {
  return useSWRMutation('updateSiteConfig', sendRequest /* options */);
}

export function useAddComment() {
  return useSWRMutation('createPost', sendRequest /* options */);
}

export function useSignIn() {
  return useSWRMutation('signIn', sendRequest /* options */);
}

export function useSignUp() {
  return useSWRMutation('signUp', sendRequest /* options */);
}

export function useUpdateUser() {
  return useSWRMutation('updateUser', sendRequest /* options */);
}

export function useAppConfig() {
  return useWrappedSWR('getAppConfig', { fallbackData: appConfigInitalState });
}

export function useAppConfigACP() {
  return useWrappedSWR('getAppConfigACP', { fallbackData: appConfigInitalState });
}

export function useTranslation() {
  return useWrappedSWR('getTranslations', { fallbackData: languageInitalState});
}

export function useComments(currentPage: number) {
  return useWrappedSWR(['loadCommentsFromServer', currentPage]);
}

export function useSearchResult(keyword: string) {
  return useWrappedSWR(['search', keyword]);
}

export function useCommentsList({
  isSearch,
  keyword,
  currentPage
}: {
  isSearch: boolean;
  keyword: string,
  currentPage: number
}) {
  return useWrappedSWR(isSearch ? ['search', keyword] : ['loadCommentsFromServer', currentPage], {
    fallbackData: commentListInitalState
  });
}

export function useLogoutUser() {
  return useSWRMutation('signOut', fetchByFunctionName);
}