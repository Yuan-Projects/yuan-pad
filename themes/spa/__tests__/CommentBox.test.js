import renderer from 'react-test-renderer';
import { useAppConfig, useTranslation, useAddComment, useUser } from '../src/common/dataHooks';
import CommentBox from '../src/front/commentBox';
import translations from '../__mocks__/site-translate-en';
import config from '../__mocks__/app-config1';
import userGuest from '../__mocks__/user-guest';

jest.mock('../src/common/dataHooks', () => ({
  useAppConfig: jest.fn(() => ({
    data: {},
    error: undefined,
  })),
  useTranslation: jest.fn(() => ({
    data: {},
    error: undefined,
  })),
  useUser: jest.fn(() => ({
    user: {},
    error: undefined,
  })),
  useAddComment: jest.fn(() => ({
    trigger: () => {},
  })),
}));

it('CommentBox renders correctly', () => {
  (useTranslation).mockImplementation(() => ({
    data: translations,
    loadingError: undefined,
  }));
  useAppConfig.mockImplementation(() => ({
    data: config,
    loadingError: undefined,
  }));
  useUser.mockImplementation(() => ({
    user: userGuest,
    isError: undefined,
  }));
  (useAddComment).mockImplementation(() => ({
    trigger: () => {}
  }));
  const comments = [
    {
        "id": 3,
        "ip": "::1",
        "uid": null,
        "uname": "admin",
        "post_content": "abc",
        "time": "2022-12-05 22:44",
        "reply_id": null,
        "reply_content": "",
        "reply_time": "01-01 08:00",
        "b_username": null
    },
    {
        "id": 2,
        "ip": "::1",
        "uid": null,
        "uname": "admin",
        "post_content": "k",
        "time": "2022-12-05 16:02",
        "reply_id": "1",
        "reply_content": "k reply",
        "reply_time": "12-05 16:22",
        "b_username": null
    },
    {
        "id": 1,
        "ip": "::1",
        "uid": null,
        "uname": "rainyjune",
        "post_content": "Welcome to our site.",
        "time": "2022-12-05 15:57",
        "reply_id": null,
        "reply_content": "",
        "reply_time": "01-01 08:00",
        "b_username": null
    }
  ];
  const component = renderer.create(<CommentBox
      onCommentCreated={() => {}}
      onPageChanged={(n) => {}}
      onCloseSearch={() => {}}
      searchText=''
      isSearch={false}
      commentsData={ {
        currentPage: 1,
        commentListType: 1,
        comments: comments,
        commentsTotalNumber: comments.length
      }}
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});