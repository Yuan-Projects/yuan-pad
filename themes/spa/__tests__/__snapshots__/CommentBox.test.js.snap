// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox renders correctly 1`] = `
<div
  className="commentBox"
>
  <h1>
    Welcome to Our site
  </h1>
  <div
    className="commentList"
  >
    <div
      className="comment"
    >
      <span
        className="commentAuthor"
        dangerouslySetInnerHTML={
          {
            "__html": "admin",
          }
        }
      />
      <span
        className="commentDate"
      >
        2022-12-05 22:44
      </span>
      <div
        className="commentText"
      >
        <p
          dangerouslySetInnerHTML={
            {
              "__html": "abc",
            }
          }
        />
      </div>
    </div>
    <div
      className="comment"
    >
      <span
        className="commentAuthor"
        dangerouslySetInnerHTML={
          {
            "__html": "admin",
          }
        }
      />
      <span
        className="commentDate"
      >
        2022-12-05 16:02
      </span>
      <div
        className="commentText"
      >
        <p
          dangerouslySetInnerHTML={
            {
              "__html": "k",
            }
          }
        />
      </div>
      <div
        className="reply"
        dangerouslySetInnerHTML={
          {
            "__html": "<font color="red"> undefined replied at 12-05 16:22 :</font> k reply",
          }
        }
      />
    </div>
    <div
      className="comment"
    >
      <span
        className="commentAuthor"
        dangerouslySetInnerHTML={
          {
            "__html": "rainyjune",
          }
        }
      />
      <span
        className="commentDate"
      >
        2022-12-05 15:57
      </span>
      <div
        className="commentText"
      >
        <p
          dangerouslySetInnerHTML={
            {
              "__html": "Welcome to our site.",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="statistics"
  >
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "We have <span id="totalNum"> 3 </span> post(s), <span id="totalPages"> 1 </span> page(s)",
        }
      }
    />
  </div>
  <form
    className="commentForm form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="col-sm-2 col-lg-2 control-label"
        htmlFor="inputUser"
      >
        NickName
      </label>
      <div
        className="col-sm-5 col-lg-5"
      >
        <input
          className="form-control"
          id="inputUser"
          maxLength="10"
          onChange={[Function]}
          type="text"
          value="anonymous"
        />
        <label
          className="control-label"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="col-sm-2 col-lg-2 control-label"
        htmlFor="inputContent"
      >
        Content
      </label>
      <div
        className="col-sm-10 col-lg-10"
      >
        <textarea
          className="form-control"
          id="inputContent"
          onChange={[Function]}
          rows="3"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-sm-offset-2 col-sm-10 col-lg-offset-2 col-lg-10"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
